state_machine/applications/system-check/tests/coil_test.py

```diff
@@ -0,0 +1,137 @@
+"""
+Python system check script for PyCubed satellite board
+PyCubed Mini mainboard-v02 for Pocketqube Mission
+Torque Driver Test
+* Author(s): Yashika Batra
+"""
+
+import time
+
+# voltage level constants; set between 0 and 1
+# get these values from the hex values in doc
+v1 = 0x0A
+v2 = 0x15
+v3 = 0x1F
+
+def user_test(driver, coilidx):
+    """
+    All user interaction happens in this function
+    Set wait times, change print statement and input formatting, etc.
+    """
+
+    # wait times
+    wait_time = 30
+    driver_time = 10
+
+    # formula in drv8830 docs:
+    # https://www.ti.com/lit/ds/symlink/drv8830.pdf?ts=1657017752384&ref_url=https%253A%252F%252Fwww.ti.com%252Fproduct%252FDRV8830
+    projected_voltage1 = 4 * 1.285 * int(v1) / 64
+    projected_voltage2 = 4 * 1.285 * int(v2) / 64
+    projected_voltage3 = 4 * 1.285 * int(v3) / 64
+
+    # set up the test
+    print("Testing Coil", str(coilidx), "for the following voltage levels: " +
+          str(projected_voltage1) + ", " + str(projected_voltage2 + ", " +
+          str(projected_voltage3)))
+    print("Please retrieve a magnetometer. A cellphone app will also work. " +
+          "Waiting", str(wait_time), "seconds.")
+    time.sleep(wait_time)
+
+    # figure out what pins we need to be reading
+    # conduct the test
+    print("Beginning the test now. Please note the magnetometer readings for the next",
+          str(driver_time * 3), "seconds.")
+
+    # test each voltage level
+    driver.vout(v1)
+    time.sleep(driver_time)
+    driver.vout(v2)
+    time.sleep(driver_time)
+    driver.vout(v3)
+    time.sleep(driver_time)
+
+    # check user input
+    print("Data Collection Complete")
+    success = input("Did the magnetometer readings increase over the last " +
+                    str(driver_time * 3) + " seconds? (Y/N): ")
+
+    return success
+
+
+def voltage_levelx(cubesat, result_dict, coilidx):
```

Comment:
My general preference is to not abbreviate and use underscores to separate words. `dict` is probably ok, as it is a very common abbreviation and significantly shortens the word (though I still think `dictionary` is better), but I would use `coil_index` - it's much more readable than `coilidx`.